### ISSUE TITLE
App Router Compatibility

### ISSUE THREAD

**Issue by colin27091** [premieroctet/next-admin#54]
Setting up compatibility for the `App Router`

**Comment by avrohom-mada** [comment]
Tell me what it takes maybe I'll try to set it up.

**Comment by vincentwinkel** [comment]
+1

**Comment by colin27091** [comment]
We have been working on this functionality for a few weeks.

The sticking point is that the page.ts file doesn't access the full request, and to feed our routing system we need the full request.

One solution would be to create an API endpoint (route.ts) to get the full request, but this solution requires us to have two routes, one for the page.tsx and one for the API endpoint, but we want to keep a minimal structure and create only one route.

The most likely theoretical approach would be to use server actions in page.tsx. By providing a server action, developers could bind it into their page and use it in their <NextAdmin> component.

We could imagine an integration like this (non-contractual code ^^):

```tsx
// app/admin/[[...nextadmin]]/page.tsx
`use client`;

import { NextAdmin, nextAdminServerAction } from "@premieroctet/next-admin";

export const AdminComponent = () => {
  return <NextAdmin action={nextAdminServerAction} .../>;
}
```
Implementing such a system would require us to change our current structure, and even more work to support both the Page Router and the App Router.

However, you can try a POC of this new architecture. We will probably work on this before the end of the year.

**Comment by foyarash** [comment]
App Router support has been released in 3.0.0
